pkgs/development/libraries/opencascade/default.nix

```diff
@@ -33,6 +44,10 @@ stdenv.mkDerivation rec {
       url = "https://github.com/tpaviot/oce/commit/3b44656e93270d782009b06ec4be84d2a13f8126.patch";
       sha256 = "1ccakkcwy5g0184m23x0mnh22i0lk45xm8kgiv5z3pl7nh35dh8k";
     })
+    (fetchpatch {
+      url = "https://github.com/tpaviot/oce/commit/cf50d078cd5fac03a48fd204938bd240930a08dc.patch";
+      sha256 = "1szrh4ai9dfkwhk425glyz4brrfhw1vg06jcsnmq18fgmih6x6kj";
```

Comment:
I get:
```
hash mismatch in fixed-output derivation '/nix/store/an2f9fnbn5p0yzr0yxja57jgs8kjhzqa-cf50d078cd5fac03a48fd204938bd240930a08dc.patch':
  wanted: sha256:1szrh4ai9dfkwhk425glyz4brrfhw1vg06jcsnmq18fgmih6x6kj
  got:    sha256:1xv94hcvggmb1c8vqwic1aiw9jw1sxk8mqbaak9xs9ycfqdvgdyc
```

Reply:
Ok, interesting. I ran

```
hannesweisbach@MBP nixpkgs % nix-prefetch-url --type sha256 http://github.com/tpaviot/oce/commit/cf50d078cd5fac03a48fd204938bd240930a08dc.patch
[0.0 MiB DL]
path is '/nix/store/cdqknhxvd9p2g0yzm6kdxpr9s3k90syl-cf50d078cd5fac03a48fd204938bd240930a08dc.patch'
1szrh4ai9dfkwhk425glyz4brrfhw1vg06jcsnmq18fgmih6x6kj
```

Is this incorrect?

Reply:
`fetchpatch` cleans up the patch after downloading, so the hash you get is different from `fetchurl`.

Reply:
Ah, I understand. So, because I executed `nix-prefetch-url` it didn't clean up the patch and showed me the "wrong" hash, but I was able to install it because during install the prefetched patch got used (i.e. ´fetchpatch` never ran for this patch).

That's funky … and unexpected.

Reply:
Fixed.

Btw: What's the canonical way to get a hash within nix?

Reply:
> Btw: What's the canonical way to get a hash within nix?

Since this was a hash of a fixed-output derivation, the only generic way to see what the result will be is to attempt building it.

Reply:
I think the most common way for `fetchpatch` is to build using a wrong hash and look at the error message.  There's even a canonical wrong hash: `lib.fakeSha256`

With `src` attributes, you can usually do `nix-prefetch-url -A opencascade.src`